src/cassette/TsxImage.cc

```diff
@@ -0,0 +1,480 @@
+				cliComm.printInfo("Block#13");
+#endif
+				pos += writeBlock13((Block13*)&buf[pos]);
+			} else
+			if (bid == 0x20) {	//Silence Block
+#ifdef DEBUG
+				cliComm.printInfo("Block#20");
+#endif
+				pos += writeBlock20((Block20*)&buf[pos]);
+			} else
+			if (bid == 0x30) {	//Text description Block
+#ifdef DEBUG
+				cliComm.printInfo("Block#30");
+#endif
+				pos += writeBlock30((Block30*)&buf[pos], cliComm);
+			} else
+			if (bid == 0x32) {	//Archive info Block
+#ifdef DEBUG
+				cliComm.printInfo("Block#32");
+#endif
+				pos += writeBlock32((Block32*)&buf[pos], cliComm);
+			} else
+			if (bid == 0x35) {	//Custom info Block
+#ifdef DEBUG
+				cliComm.printInfo("Block#35");
+#endif
+				pos += writeBlock35((Block35*)&buf[pos]);
+			} else
+			if (bid == 0x4B) {	//MSX KCS Block
+				Block4B *b = (Block4B*)&buf[pos];
+#ifdef DEBUG
+				cliComm.printInfo("Block#4B");
+#endif
+				//determine file type
+				if (firstFile && (pos+12+5)<size && b->blockLen-12==16) {
+					FileType type = CassetteImage::UNKNOWN;
+					if (!memcmp(&(b->data), ASCII_HEADER, 10)) {
+						type = CassetteImage::ASCII;
+					} else if (!memcmp(&(b->data), BINARY_HEADER, 10)) {
+						type = CassetteImage::BINARY;
+					} else if (!memcmp(&(b->data), BASIC_HEADER, 10)) {
+						type = CassetteImage::BASIC;
+					}
+					setFirstFileType(type);
+					firstFile = false;
+				}
+				//read the block
+				pos += writeBlock4B(b);
+			} else
+			if (bid == 0x5A) {	//Glue Block
+#ifdef DEBUG
+				cliComm.printInfo("Block#5A");
+#endif
+				pos += 10;
+			} else
+			if (bid == 0x14) {	//Pure data Block
+				cliComm.printWarning("Block#14 Unsupported yet!");
+				pos += *((uint24_t*)&buf[pos+8]) + 11;
+			} else
+			if (bid == 0x15) {	//Direct recording Block
+				cliComm.printWarning("Block#15 Unsupported yet!");
+				pos += *((uint24_t*)&buf[pos+6]) + 9;
+			} else
+			if (bid == 0x19) {	//Generalized data block
+				cliComm.printWarning("Block#19 Unsupported yet!");
+				pos += *((uint32_t*)&buf[pos+1]) + 5;
+			} else
+			if (bid == 0x21) {	//Group start Block
+				cliComm.printWarning("Block#21 Unsupported yet!");
+				pos += *((uint8_t*)&buf[pos+1]) + 1;
+			} else
+			if (bid == 0x22) {	//Group end Block
+				cliComm.printWarning("Block#22 Unsupported yet!");
+				pos += 1;
+			} else
+			if (bid == 0x23) {	//Jump Block
+				cliComm.printWarning("Block#23 Unsupported yet!");
+				pos += 3;
+			} else
+			if (bid == 0x24) {	//Loop start Block
+				cliComm.printWarning("Block#24 Unsupported yet!");
+				pos += 3;
+			} else
+			if (bid == 0x25) {	//Loop end Block
+				cliComm.printWarning("Block#25 Unsupported yet!");
+				pos += 1;
+			} else
+			if (bid == 0x26) {	//Call sequence Block
+				cliComm.printWarning("Block#26 Unsupported yet!");
+				pos += *((uint16_t*)&buf[pos+1])*2 + 3;
+			} else
+			if (bid == 0x27) {	//Return sequence Block
+				cliComm.printWarning("Block#27 Unsupported yet!");
+				pos += 1;
+			} else
+			if (bid == 0x31) {	//Message Block
```

Comment:
All magic numbers used in this _if_ statements could be declared as constants. It would be clearer, less error-prone and would make the comments pointless.